node/core/runtime-api/src/lib.rs

```diff
@@ -36,22 +36,38 @@ use polkadot_primitives::v1::{Block, BlockId, Hash, ParachainHost};
 use sp_api::ProvideRuntimeApi;
 use sp_core::traits::SpawnNamed;
 
-use futures::prelude::*;
-use std::sync::Arc;
+use futures::{prelude::*, stream::FuturesUnordered, channel::oneshot, select};
+use std::{sync::Arc, collections::VecDeque, pin::Pin};
 
 const LOG_TARGET: &str = "runtime_api";
 
+/// The number of maximum parallel runtime api requests.
+const MAX_PARALLEL_REQUESTS: usize = 2;
```

Comment:
I think we can do more and need more. For example, during backing there is a place where we start 4 runtime API requests at the same time. Maybe we should do 4?